### PR TITLE
KOGITO-5413 Upgrade minimum Java release to 11

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: optaplanner-core/src/main/java/org/optaplanner/core/api/domain/package-info.java [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-/**
- * Domain annotations and support classes.
- */
-package org.optaplanner.core.api.domain;
```
Comment (ge0ffrey) [inline optaplanner-core/src/main/java/org/optaplanner/core/api/domain/package-info.java:20]: Let's not remove the package javadocs.

File: optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc [hunk 2]
```diff
@@ -33,8 +33,8 @@ that deliver a near-optimal solution in a reasonable amount of time.
 
 == What you'll need
 
-* https://www.oracle.com/java/technologies/javase-downloads.html[JDK 8 or later]
-* https://maven.apache.org/download.cgi[Maven 3.2+] or https://gradle.org/install/[Gradle4+]
+* https://jdk.java.net/[JDK 11 or later]
```
Comment (ge0ffrey) [inline optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc:36]: Let's link to https://adoptopenjdk.net/ like we do in the quickstart readme's.

Motivation: Oracle's JDK downloads are non-free or free-for-a-limited-period AFAIK.
The AdoptOpenJDK project is where the community is working around that.

Note: AdoptOpenJDK is in the process of being renamed to Adoptium, but we will only update our links from adoptopenjdk.net to adoptium.net once that renaming is complete. Meanwhile, we need to link to https://adoptopenjdk.net/ so we can find-in-path it easily later this year.

**Review by ge0ffrey — COMMENTED** [review]
Good to merge if both comments (package javadocs and link) are addressed (both non-optional).

**Comment by triceo** [comment]
Apps failure unrelated.

**Pull request merged**